adobe/spectrum-web-components
# Color controls focus management updates.

## Description
- prevent `:focus` delivery on `sp-color-handle` both at the Handle level in CSS and in the usage level by removing `tabindex=-1`.
- ensure `Arrow*` keypresses activate `focused` delivery
- remove `modifier` support for `Meta` key as it prevent `keyup` events

## Related issue(s)
fixes #1701
fixes #1698

## Motivation and context
- correct visual delivery of the current focus state
- prevent functional incongruities when keydown states can't be ended

## How has this been tested?

    1. Go to https://color-handle--spectrum-web-components.netlify.app/storybook/?path=/story/color-area--default
    2. Click the label for the `Theme` picker along the bottom
    3. Press `Shift + Tab`
    4. Ensure that the handle has enlarged.
    5. Press any of the `Arrow*` keys
    6. Ensure that the handle/value has moved/been updated
    7. Press `Shift + Tab`
    8. Ensure that after dragging the handle has returned to its original size.
    9. Press any of the `Arrow*` keys
    10. Ensure that the handle/value has not moved/been updated
    11. Press `Shift`
    12. Ensure that the handle has enlarged.
    13. Press any of the `Arrow*` keys
    14. Ensure that the handle/value has moved/been updated
    15. NOTE: Safari is particularly picky about this functionality due to how it assigns elements as "keyboard focusable" or not.
    16. REPEAT for:
      - https://color-handle--spectrum-web-components.netlify.app/storybook/?path=/story/color-slider--default
      - https://color-handle--spectrum-web-components.netlify.app/storybook/?path=/story/color-wheel--default
    1. Go to https://color-handle--spectrum-web-components.netlify.app/storybook/?path=/story/color-area--default
    2. Click and drag the Color Handle
    3. Ensure while dragging that the handle is enlarged.
    4. Ensure that after dragging the handle has returned to its original size.
    5. Ensure that the handle does not exhibit a "focus ring" of any kind.
    6. Press any of the `Arrow*` keys
    7. Ensure that the handle has enlarged
    8. Ensure that the handle/value has moved/been updated
    15. NOTE: Safari is particularly picky about this functionality due to how it assigns elements as "keyboard focusable" or not.

## Types of changes

## Checklist

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
